tests/test_pillow.py

```diff
@@ -124,6 +125,72 @@ def test_set_background_color_rgb_color_argument_check(self):
             str(e.exception), "the 'color' argument must be a 3-element tuple or list"
         )
 
+    def test_transform_colorspace_to_srgb_noop(self):
+        with open("tests/images/flower.jpg", "rb") as f:
+            image = PillowImage.open(JPEGImageFile(f))
+
+        transformed_image = image.transform_colorspace_to_srgb()
+
+        # Statistics about color values should be the same - it should be a no-op after all
+        stat = ImageStat.Stat(image.image)
+        stat_transformed = ImageStat.Stat(transformed_image.image)
+        self.assertEqual(stat.sum, stat_transformed.sum)
+
+        self.assertEqual(transformed_image.image.mode, "RGB")
+
+    def test_transform_colorspace_to_srgb_preserve_transparency(self):
+        with open("tests/images/transparent_with_icc_profile.png", "rb") as f:
+            image = PillowImage.open(PNGImageFile(f))
+
+        # The sample image is in RGBA mode, it contains transparent pixels
+        self.assertEqual(image.image.mode, "RGBA")
+
+        transformed_image = image.transform_colorspace_to_srgb()
+        # Image remains in RGBA mode
+        self.assertEqual(transformed_image.image.mode, "RGBA")
+
+        # Check that the alpha of pixel 1,1 is 0
+        self.assertEqual(transformed_image.image.convert("RGBA").getpixel((1, 1))[3], 0)
+
+    def test_transform_colorspace_to_srgb(self):
+        with open("tests/images/dog_and_lake_cmyk_with_icc_profile.jpg", "rb") as f:
+            image = PillowImage.open(JPEGImageFile(f))
+
+        # The sample image should originally be in CMYK mode
+        self.assertEqual(image.image.mode, "CMYK")
+        self.assertIsNotNone(image.get_icc_profile())
+        cms_profile = ImageCms.ImageCmsProfile(io.BytesIO(image.get_icc_profile()))
+
+        # The original embedded profile should be called "ISO Coated v2 (built-in)"
+        self.assertEqual(
+            cms_profile.profile.profile_description, "ISO Coated v2 (built-in)"
+        )
+
+        image_srgb = image.transform_colorspace_to_srgb()
+
+        # The image should now be in RGB mode as a result of the operation
+        self.assertEqual(image_srgb.image.mode, "RGB")
+
+        # We verify the result by comparing the squared sum of the color channels
+        # to known good values. This should be accurate enough to detect any major
+        # deviations from the expected result without being too sensitive to minor
+        # differences between systems.
+        stat = ImageStat.Stat(image_srgb.image)
+        expected_sum2 = [1286116001.0, 1222827100.0, 1144012271.0]
+
+        for actual, expected in zip(stat.sum2, expected_sum2):
+            self.assertAlmostEqual(
+                actual,
+                expected,
+                delta=1000,
```

Comment:
I've assumed that the results on my system might not be the same as on other systems. Hence the the `assertAlmostEqual`.

I haven't actually run this test on any other systems so not sure if my assumption is actually true.

Reply:
Let's try to be more strict and see what the CI says

Reply:
CI is green with no delta, I'll make this much simpler.